code/modules/food/beverages/bottler.dm

```diff
@@ -0,0 +1,413 @@
+		return 1
+	if(!user.canUnEquip(O, 0))
+		to_chat(user, "<span class='warning'>[O] is stuck to your hand, you can't seem to put it down!</span>")
+		return 0
+	if(is_type_in_list(O,acceptable_items))
+		if(istype(O, /obj/item/weapon/reagent_containers/food/snacks))
+			var/obj/item/weapon/reagent_containers/food/snacks/S = O
+			user.unEquip(S)
+			if(S.reagents && !S.reagents.total_volume)		//This prevents us from using empty foods, should one occur due to some sort of error
+				to_chat(user, "<span class='warning'>[S] is gone, oh no!</span>")
+				qdel(S)			//Delete the food object because it is useless even as food due to the lack of reagents
+			else
+				insert_item(S, user)
+			return 1
+		else if(istype(O, /obj/item/weapon/reagent_containers/food/drinks/cans))
+			var/obj/item/weapon/reagent_containers/food/drinks/cans/C = O
+			if(C.reagents)
+				if(C.canopened && C.reagents.total_volume)		//This prevents us from using opened cans that still have something in them
+					to_chat(user, "<span class='warning'>Only unopened cans and bottles can be processed to ensure product integrity.</span>")
+					return 0
+				user.unEquip(C)
+				if(!C.reagents.total_volume)		//Empty cans get recycled, even if they have somehow remained unopened due to some sort of error
+					recycle_container(C)
+				else								//Full cans that are unopened get inserted for processing as ingredients
+					insert_item(C, user)
+			return 1
+		else
+			user.unEquip(O)
+			insert_item(O, user)
+			return 1
+	else if(istype(O, /obj/item/trash/can))			//Crushed cans (and bottles) are returnable still
+		var/obj/item/trash/can/C = O
+		user.unEquip(C)
+		recycle_container(C)
+		return 1
+	else if(istype(O, /obj/item/stack/sheet))		//Sheets of materials can replenish the machine's supply of drink containers (when people inevitably don't return them)
+		var/obj/item/stack/sheet/S = O
+		user.unEquip(S)
+		process_sheets(S)
+		return 1
+	else		//If it doesn't qualify in the above checks, we don't want it. Inform the person so they (ideally) stop trying to put the nuke disc in.
+		to_chat(user, "<span class='warning'>You aren't sure this is able to be processed by the machine.</span>")
+		return 0
+	//..()
+
+/obj/machinery/bottler/proc/insert_item(obj/item/O, mob/user)
+	if(!O || !user)
+		return
+	if(slots[1] && slots[2] && slots[3])
+		to_chat(user, "<span class='warning'>[src] is full, please remove or process the contents first.</span>")
+		return
+	var/slot_inserted = 0
+	for(var/i = 1, i <= slots.len, i++)
+		if(!slots[i])
+			slots[i] = O
+			slot_inserted = i
+			break
+	if(!slot_inserted)
+		to_chat(user, "<span class='warning'>Something went wrong and the machine spits out [O].</span>")
+		O.forceMove(loc)
+	else
+		to_chat(user, "<span class='notice'>You load [O] into the [slot_inserted]\th ingredient tray.</span>")
+		O.forceMove(src)
+	updateUsrDialog()
+
+/obj/machinery/bottler/proc/eject_items(var/slot = 4)
+	var/obj/item/O = null
+	if(slot == 4)
```

Comment:
"Slot 4 means everything" is a magic number. May as well just use 0/null and not have to worry about specifying a default in the proc arguments.